ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-segment-security.j2

```diff
@@ -0,0 +1,64 @@
+{#
+ Copyright (c) 2023-2024 Arista Networks, Inc.
+ Use of this source code is governed by the Apache License 2.0
+ that can be found in the LICENSE file.
+#}
+{# eos - router segment-security #}
+{% if router_segment_security is arista.avd.defined %}
+!
+router segment-security
+{%     if router_segment_security.enabled is arista.avd.defined(true) %}
+   no shutdown
+{%     endif %}
+{%     if router_segment_security.policies is arista.avd.defined %}
+{%         for policy in router_segment_security.policies %}
+   !
+   policy {{ policy.name }}
+{%             for sequence in policy.sequence_numbers | arista.avd.natural_sort('sequence') %}
+{%                 if sequence.action == "redirect" %}
+{%                     set eos_cli = sequence.action ~ " next-hop " ~ sequence.next_hop %}
+{%                 else %}
+{%                     set eos_cli = sequence.action %}
+{%                 endif %}
+{%                 if sequence.stateless | arista.avd.default (true) %}
+{%                     set eos_cli = eos_cli ~ " stateless" %}
+{%                 endif %}
+{%                 if sequence.log is arista.avd.defined(true) %}
+{%                     set eos_cli = eos_cli ~ " log" %}
+{%                 endif %}
+      {{ sequence.sequence }} application {{ sequence.application }} action {{ eos_cli }}
+{%             endfor %}
+{%         endfor %}
+{%     endif %}
+{%     if router_segment_security.vrfs is arista.avd.defined %}
+{%         for vrf in router_segment_security.vrfs | arista.avd.natural_sort('name') %}
+   !
+   vrf {{ vrf.name }}
+{%             for segment in vrf.segments %}
+      segment {{ segment.name }}
+         definition
+{%                 if segment.definition.interfaces is arista.avd.defined %}
+{%                     for entry in segment.definition.interfaces | arista.avd.natural_sort %}
+            match interface {{ entry }}
+{%                     endfor %}
+{%                 endif %}
```

Comment:
Natural sort contains the defined check and will return an empty list if not set.
```suggestion
{%                 for entry in segment.definition.interfaces | arista.avd.natural_sort %}
            match interface {{ entry }}
{%                 endfor %}
```